go/libraries/doltcore/doltdb/durable/index.go

```diff
@@ -37,6 +40,10 @@ type Index interface {
 
 	// Format returns the types.NomsBinFormat for this index.
 	Format() *types.NomsBinFormat
+
+	// AddColumnToRows adds the column given to the rows data and returns the resulting rows.
+	// The |newCol| is present in |newSchema|.
+	AddColumnToRows(ctx context.Context, newCol string, newSchema schema.Schema) (Index, error)
```

Comment:
I think this would be better as a top-level function rather than on this interface, but it's nbd